crates/astria-conductor/src/executor/state.rs

```diff
@@ -280,9 +285,14 @@ impl State {
     pub(crate) fn try_from_genesis_info_and_commitment_state(
         genesis_info: GenesisInfo,
         commitment_state: CommitmentState,
+        commit_level: crate::config::CommitLevel,
     ) -> Result<Self, InvalidState> {
-        let _ = map_firm_to_sequencer_height(&genesis_info, &commitment_state)?;
-        let _ = map_soft_to_sequencer_height(&genesis_info, &commitment_state)?;
+        if commit_level.is_with_firm() {
+            let _ = map_firm_to_sequencer_height(&genesis_info, &commitment_state)?;
+        }
+        if commit_level.is_with_soft() {
```

Comment:
This condition is wrong. Soft must update with firm in lock-step even if soft is not set.